File: 1784-check-if-binary-string-has-at-most-one-segment-of-ones/1784-check-if-binary-string-has-at-most-one-segment-of-ones.cpp
class Solution {
public:
    bool checkOnesSegment(string s) {
        bool ans =false;
        bool zeroes =false;
        for(int i=0;i<s.size();i++){
            if(s[i]=='1'){
                ans= true;
            }
            if(s[i]=='0'){
                if(ans==true) zeroes =true;
            }
            if(zeroes && s[i]=='1') return false;
        }
        return ans;
        
    }
};

File: 1784-check-if-binary-string-has-at-most-one-segment-of-ones/1784-check-if-binary-string-has-at-most-one-segment-of-ones.java
class Solution {
    public boolean checkOnesSegment(String s) {
        boolean ans = false;
        boolean zeroes = false;
        for(int i=0;i<s.length();i++){
            if(s.charAt(i)=='1'){
                if(zeroes) return false;
                ans=true;
            }
            else{
                if(ans) zeroes=true;
            }
        }
        return ans;
        
    }
}